packages/zowe-explorer/src/trees/job/ZoweJobNode.ts

```diff
@@ -420,6 +421,10 @@ export class ZoweSpoolNode extends ZoweJobNode {
         this.spool = opts.spool;
         const icon = IconGenerator.getIconByNode(this);
 
+        // parent of parent should be the session; tie resourceUri with TreeItem for file decorator
+        if (opts.spool && opts.parentNode && opts.parentNode.getParent()) {
+            this.resourceUri = JobSpoolProvider.encodeJobFile(opts.parentNode.getParent().label as string, opts.spool);
+        }
```

Comment:
I'm wondering if we want to consolidate the assignments to `resourceUri`? We set it on lines 417-419 as long as `parentNode` is valid. I think we'll want to keep that assignment and maybe enhance it so that we only set it when `opts.spool` is defined.